### PR TITLE
Fix #654: time-aware log-return Sharpe replaces daily-assuming sqrt(252)

### LINKED ISSUE #654: [ERROR] Backtest Sharpe misannualized: reports +1.11 on a strategy that lost 14% with 72% drawdown
`calculate_sharpe` (src/gimmes/reporting/metrics.py:49-68) annualizes by sqrt(252) assuming DAILY returns, but backtest/report.py:40-49 feeds it per-SETTLEMENT equity steps (61 steps over ~2.5 months). Combined with variance drag (mean simple step-return is positive at +0.073/std while compounded total is -14%), the Jan-Jul 2026 backtest reports **Sharpe 1.11 alongside net -$1,403.93 (-14% ROI) and 71.8% max drawdown** — a number that could greenlight a losing live deployment.

Hand-verified: per-trade Sharpe = -0.037; equity-step mean/std = +0.073; 0.073 x sqrt(252) ≈ 1.16 ≈ reported 1.11.

**Fix direction:** compute Sharpe on time-normalized returns (e.g., resample the equity curve to daily using settle timestamps) or on log returns with the correct periods-per-year for the actual observation frequency; at minimum label the current metric per-trade and drop the sqrt(252).

### PR DESCRIPTION
**Pull request by jonnymobs907**
## Summary

The backtest reported **Sharpe 1.11 on a strategy that lost 14% with a 72% drawdown** — and the Scorecard agent rates health HEALTHY at ≥ 1.0 off this number. Two compounding defects in `calculate_sharpe`: it annualized any series by √252 as if daily (the backtest feeds ~61 per-settlement steps over 6 months → ~2× inflation), and arithmetic-mean simple returns hide variance drag (mean step return positive while compounded growth negative).

`calculate_sharpe_from_curve` fixes both: **log returns** (sign telescopes to the compounded total — a loser can't print a positive Sharpe) annualized by **observed frequency** derived from the curve's real time span. Both callers migrated (backtest + live scorecard); the daily-assuming function is removed entirely. Naive timestamps normalize to UTC (previously an uncaught TypeError past the parse guard).

## Consumer note
`summary.sharpe` / `sharpe_ratio` keys unchanged, but **values drop substantially** — historical scorecards are not comparable, and Scorecard health ratings may transition on the next run. That is the correction working.

## Review
Combined review + mutation testing: sqrt(252)-restore, log→simple, and ddof mutations all killed (the ddof kill required de-degenerating the hand-computed pin — a symmetric curve pins nothing); sign-identity docstring corrected to its precise scope (holds when all equity points positive); rf-conversion approximation documented (exact at the rf=0 default). Simplifier applied.

## Test plan
- `uv run pytest tests/unit/test_metrics.py tests/unit/test_backtest_report.py` — annualization-by-observed-frequency pin (daily vs weekly = √7), variance-drag negative pin, hand-computed value, sign-matches-ROI, dropped-pair arithmetic, edge cases.
- Full suite green; ruff clean.

Closes #654

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01XraN34AP4re87cAzt9LRKB

### CODE REVIEW

File: src/gimmes/reporting/metrics.py [hunk 1]
```diff
@@ -46,26 +47,79 @@ def calculate_max_drawdown(equity_curve: list[float]) -> tuple[float, float]:
...
+    log_returns: list[float] = []
+    for (_, prev), (_, cur) in zip(curve, curve[1:], strict=False):
+        if prev > 0 and cur > 0:
+            log_returns.append(math.log(cur / prev))
+    if len(log_returns) < 2:
+        return 0.0
+
+    try:
+        t0 = datetime.fromisoformat(curve[0][0])
+        t1 = datetime.fromisoformat(curve[-1][0])
```
[the remark this answers is not in the document]
Comment (jonnymobs907) [reply src/gimmes/reporting/metrics.py:100]: This is handled correctly on this repo's runtime — Python 3.11+ `datetime.fromisoformat` parses `Z`-suffixed strings natively (verified on the project's 3.13: `fromisoformat('2026-01-01T00:00:00Z')` → aware datetime), so no ValueError/0.0 masking occurs. The `replace('Z', '+00:00')` pattern at fee_cache.py:34 predates the 3.11 parser upgrade. Whitespace-padded timestamps do reject, but every producer feeding this function emits `.isoformat()` or SQLite `datetime('now')` output — no padded path exists.

File: src/gimmes/reporting/metrics.py [hunk 2]
```diff
@@ -46,26 +47,79 @@ def calculate_max_drawdown(equity_curve: list[float]) -> tuple[float, float]:
...
+    zero variance, non-positive equity pairs, unparseable timestamps,
+    or a non-positive time span.
     """
-    if len(returns) < 2:
+    if len(curve) < 2:
         return 0.0
 
-    excess = [r - risk_free_rate for r in returns]
+    log_returns: list[float] = []
+    for (_, prev), (_, cur) in zip(curve, curve[1:], strict=False):
```
[the remark this answers is not in the document]
Comment (jonnymobs907) [reply src/gimmes/reporting/metrics.py:92]: Deliberate: `strict=False` documents that the length mismatch between `curve` and `curve[1:]` is intentional (pairwise iteration), per the simplifier pass's explicit reasoning. It's semantically identical to the default, kept as intent documentation.
